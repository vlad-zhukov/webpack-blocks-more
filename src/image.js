/*
 * Image loader.
 */
export default function image(options = {}) {
    return (context, {addLoader}) =>
        addLoader({
            ...context.match,
            use: [
                {
                    loader: 'image-webpack-loader',
                    options: {
                        svgo: {
                            plugins: [
                                {
                                    removeEmptyAttrs: true,
                                },
                                {
                                    cleanupAttrs: true,
                                },
                                {
                                    removeComments: true,
                                },
                            ],
                        },
                        optipng: {
                            optimizationLevel: 5,
                        },
                        pngquant: {
                            quality: '100',
                        },
                        mozjpeg: {
                            quality: 100,
                        },
                        ...options,
                    },
                },
            ],
        });
}
