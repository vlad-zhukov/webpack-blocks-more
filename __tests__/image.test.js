import {createConfig, match} from '@webpack-blocks/webpack';
import {file} from '@webpack-blocks/assets';
import image from '../src/image';

it('should work with match()', () => {
    const config = createConfig([match(['*.jpg', '*.png'], [image()])]);

    expect(config).toMatchSnapshot();
});

it('should work with file block', () => {
    const config = createConfig([match(['*.jpg', '*.png'], [file(), image()])]);

    expect(config).toMatchSnapshot();
});
