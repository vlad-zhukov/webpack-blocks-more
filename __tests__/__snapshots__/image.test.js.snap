// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should work with file block 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "test": Array [
          /\\^\\.\\*\\\\\\.jpg\\$/,
          /\\^\\.\\*\\\\\\.png\\$/,
        ],
        "use": Array [
          Object {
            "loader": "file-loader",
            "options": Object {},
          },
          Object {
            "loader": "image-webpack-loader",
            "options": Object {
              "mozjpeg": Object {
                "quality": 100,
              },
              "optipng": Object {
                "optimizationLevel": 5,
              },
              "pngquant": Object {
                "quality": "100",
              },
              "svgo": Object {
                "plugins": Array [
                  Object {
                    "removeEmptyAttrs": true,
                  },
                  Object {
                    "cleanupAttrs": true,
                  },
                  Object {
                    "removeComments": true,
                  },
                ],
              },
            },
          },
        ],
      },
    ],
  },
  "plugins": Array [],
  "resolve": Object {
    "extensions": Array [
      ".js",
      ".json",
    ],
  },
  "stats": Object {
    "children": false,
    "chunks": false,
    "modules": false,
    "reasons": false,
  },
}
`;

exports[`should work with match() 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "test": Array [
          /\\^\\.\\*\\\\\\.jpg\\$/,
          /\\^\\.\\*\\\\\\.png\\$/,
        ],
        "use": Array [
          Object {
            "loader": "image-webpack-loader",
            "options": Object {
              "mozjpeg": Object {
                "quality": 100,
              },
              "optipng": Object {
                "optimizationLevel": 5,
              },
              "pngquant": Object {
                "quality": "100",
              },
              "svgo": Object {
                "plugins": Array [
                  Object {
                    "removeEmptyAttrs": true,
                  },
                  Object {
                    "cleanupAttrs": true,
                  },
                  Object {
                    "removeComments": true,
                  },
                ],
              },
            },
          },
        ],
      },
    ],
  },
  "plugins": Array [],
  "resolve": Object {
    "extensions": Array [
      ".js",
      ".json",
    ],
  },
  "stats": Object {
    "children": false,
    "chunks": false,
    "modules": false,
    "reasons": false,
  },
}
`;
